### PR TITLE
ramips: cosmetic changes for dts and makefile

### PR DESCRIPTION
**Pull request by mans0n**
ramips: use hex notation for *-mtd-eeprom property
ramips: use generic node name for flash
ramips: fix incorrect flash reg property
ramips: simplify palmbus/{i2c,spi} in device DTS files
ramips: tidy up image Makefile

### CODE REVIEW

**Comment by mans0n** [comment]
This PR is merely cosmetic (and it should be).
I'll explain what I've checked to avoid making obvious mistakes.

For ramips, there's two kinds of nor flash nodes: spi-nor and cfi-flash. I already know spi-nor does not care about the node name. And I tested it for cfi-flash on rt3662.
I did not touch mx25l25635f and mx25l6405d compatible as it might be intentional (c2bcdabf2a1b) and I don't know much about it.
As for the palmbus patch, I'm not sure if all of the devices actually require i2c. To me it seems to be copy & paste.
I've built netgear r6120 image to test if changing order of sercom-footer and sercom-seal does matter. And it seems it does not.

**Comment by adrianschmutzler** [comment]
One might add a justification for patch 2, e.g. from submitting patches:
"The name of a node should reflect the function of the device and not its model. Examples for generic node names can be found in Section 2.2.2 Generic Names Recommendation"
with further reference https://github.com/devicetree-org/devicetree-specification/blob/master/source/devicetree-basics.rst#generic-names-recommendation

**Comment by adrianschmutzler** [comment]
I'd merge patches 1, 4 and 5 (all except the flash-related stuff) relatively quick.

For no. 5, I'd prefer to split it into changes to image/Makefile and changes to the other image/****.mk files. Those are also different based on what they do.

This should be fairly easy to achieve with clever use of "git checkout", "git revert" and "git rebase". However, I don't know how much you like git, if it would be a pain for you I can also do it myself during merge. :-)

**Comment by mans0n** [comment]
> One might add a justification for patch 2, e.g. from submitting patches:

Okay, I'll add it.

> For no. 5, I'd prefer to split it into changes to image/Makefile and changes to the other image/****.mk files. Those are also different based on what they do.

Yeah, it was actually two, I'll split it back. I think I can manage that.

**Comment by mans0n** [comment]
Reworded flash renaming patch, and split makefile patch into two.

**Comment by mans0n** [comment]
Reworded patches and changed commit order.

**Comment by adrianschmutzler** [comment]
I've picked the mentioned four now, will do a compile-test on mt7621, and then merge them.

File: target/linux/ramips/dts/rt5350_asiarf_awm002-evb-4m.dts [hunk 1]
```diff
@@ -10,7 +10,7 @@
 &spi0 {
 	status = "okay";
 
-	m25p80: m25p80@0 {
+	m25p80: flash@0 {
```
Comment (adrianschmutzler) [inline target/linux/ramips/dts/rt5350_asiarf_awm002-evb-4m.dts:13]: That label seems to be unnecessary. Maybe drop that as a "While at it" when refreshing patches after my merge (same for the next change on 8m variant).

**Comment by adrianschmutzler** [comment]
> For ramips, there's two kinds of nor flash nodes: spi-nor and cfi-flash. I already know spi-nor does not care about the node name. And I tested it for cfi-flash on rt3662.
> I did not touch mx25l25635f and mx25l6405d compatible as it might be intentional ([c2bcdab](https://github.com/openwrt/openwrt/commit/c2bcdabf2a1b1630c644076361b243a082d36c60)) and I don't know much about it.

That is valuable information, I'd include it into the commit message of the respective patch. Maybe you can even provide a reference instead of "I already know spi-nor does not care" ...

**Comment by adrianschmutzler** [comment]
Just merged the four patches.

File: target/linux/ramips/dts/rt5350_asiarf_awm002-evb-4m.dts (hunk 1 above)
Comment (mans0n) [reply target/linux/ramips/dts/rt5350_asiarf_awm002-evb-4m.dts]: Nice catch. Removed now.

**Comment by mans0n** [comment]
Rebased and reworded.

**Comment by adrianschmutzler** [comment]
Thanks, will merge it during the day.

**Pull request closed without merging**